Utilities/Extensions/RandomExtensions.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Linq;
+
+namespace Utilities.Extensions
+{
+    public static class RandomExtensions
+    {
+        public static double[] NextVector(this Random rand, int size)
+        {
+            var vector = Enumerable.Range(0, size)
+                .Select(_ => rand.NextDouble()).ToArray();
+            var norm = vector.Norm();
+            return vector.Select(x => x / norm).ToArray();
+        }
+    }

```

Comment:
There are some extensions/utilities for vectors and matrices in #174, so we might want to wait a bit until it's merged and see what can be reused from or added there. So that we don't create a lot of methods that all do the same thing.

Reply:
Yep, in #174 there are some duplicated methods, so when it's merged, I'll rebase my changes and reuse existing logic. But I insist on replacing static methods with extension methods for vector/matrix operations and more some methods from SVD to Utilities (like **RandomUnitVector**).

Reply:
#174 is merged

Reply:
@siriak reused existing logic and covered extensions with unit tests